docs/architecture/adr-057-RPC.md

```diff
@@ -12,38 +12,57 @@ There are currently two options being discussed: gRPC & JSON-RPC.
 
 ### JSON-RPC:
 
-JSON-RPC is a json based RPC specification. Tendermint has implemented its own variant of JSON-RPC as it is not compatible with the [JSON-RPC 2.0 specification](https://www.jsonrpc.org/specification).
+JSON-RPC is a JSON-based RPC protocol. Tendermint has implemented its own variant of JSON-RPC which is not compatible with the [JSON-RPC 2.0 specification](https://www.jsonrpc.org/specification).
 
 **Pros:**
 
-- Easy to interface with (JSON)
+- Simple
+- Well-known and well-understood by users and integrators
+- Easy to interface with (HTTP and JSON)
+- Mature, common, and familiar tooling (e.g. `curl` and `jq`)
+- Integrates reasonably well with web infrastructure (proxies, API gateways, service meshes, caches, etc)
 
 **Cons:**
 
-- Transmitting JSON, which can be slower than other forms of data transmission.
-- Tendermint has its own implementation
+- No schema support
+- RPC clients must be hand-written
+- Streaming not built into protocol
+- Underspecified types (e.g. numbers and timestamps)
+- Tendermint has its own implementation (not standards compliant, maintenance overhead)
+- Stdlib `jsonrpc` package only supports JSON-RPC 1.0, no dominant package for JSON-RPC 2.0
+- Tooling around documentation/specification (e.g. Swagger) could be better
+- JSON data is larger (offset by HTTP compression)
+- Serializing is slower ([~100% marshal, ~400% unmarshal](https://github.com/alecthomas/go_serialization_benchmarks)); insignificant in absolute terms
 
 ### gRPC:
 
 gRPC is a high performant RPC framework. It has been battle tested by a large number of users and is heavily relied on and maintained by countless large corporations.
 
 **Pros:**
 
-- Efficient data retrieval for lite clients and other protocols
+- Efficient data retrieval for users, lite clients and other protocols
 - Easily implemented in supported languages (Go, Dart, JS, TS, rust, Elixir, Haskell, ...)
-- Efficient data retrieval for users
 - Can provide a REST API via [gRPC-gateway](https://github.com/grpc-ecosystem/grpc-gateway)
 - Can provide a GraphQL API via [Rejoiner](https://github.com/google/rejoiner)
-- HTTP/2-based protocol
```

Comment:
Removed this since JSON-RPC also runs on HTTP/2

Reply:
I don't believe this is true as http2 is a binary transport while http1 is plain text?

Reply:
HTTP/2 mainly changes the network transport (e.g. by multiplexing streams in the same connection), but not the HTTP protocol semantics. For example, the Golang HTTP server uses HTTP/2 by default, but developers will generally not notice any difference between the HTTP/1 or HTTP/2 server as the API is identical.

That said, it's not a given that a JSON-RPC server uses HTTP/2. It _can_, but it depends on how it is implemented.